lib/src/data/repository/movie_api_repository.dart

```diff
@@ -0,0 +1,52 @@
+import '../../core/util/data_state.dart';
+import '../../core/util/enums.dart';
+import '../../data/model/genre_model.dart';
+import '../../domain/entity/movie.dart';
+import '../../domain/repository/i_movie_repository.dart';
+import '../datasource/remote/movie_api_service.dart';
+
+class MovieApiRepository implements IMovieRepository {
+  final MovieApiService movieApiProvider = MovieApiService();
+
+  Future<void> _initGenreData() {
+    return movieApiProvider.fetchGenres();
+  }
+
+  @override
+  Future<DataState<List<Movie>>> getMovies([
+    int page = 1,
+    MovieEndpoint endpoint = MovieEndpoint.popular,
+  ]) async {
+    try {
+      if (GenreModel.validGenres.isEmpty) {
+        await _initGenreData();
+      }
```

Comment:
Why you have this logic here?

Reply:
Initially, genre data was only used to display movies properly, so it made sense to group it with the fetchMovie functionality. That way we don't need to call an extra method to get the genres before we try to display the movies, the repo already takes care of that.
However, since now the Drawer needs to know the genre data but doesn't care about the movies, I'm thinking of making a separate fetchValidGenres use case and call that before we try to fetch any movies. 
Any tips/thoughts?

Reply:
Yes, you can separate to a new UseCase to get that specific kind of movies

Reply:
done